src/index.ts

```diff
@@ -0,0 +1,208 @@
+    this.defaultProvider = _.defaultTo(
+      opts.defaultProvider,
+      LIMESTON_API_DEFAULTS.provider);
+    this.useCache = _.defaultTo(
+      opts.useCache,
+      LIMESTON_API_DEFAULTS.useCache);
+  }
+
+  // Here we can pass any async code that we need to execute on api init
+  // For example we can load provider name to address mapping here
+  static init(config: LimestoneApiConfig = {}): LimestoneApi {
+    const arweaveProxy = new ArweaveProxy();
+    const optsToCopy = _.pick(config, [
+      "defaultProvider",
+      "verifySignature",
+      "useCache",
+      "version",
+    ]);
+
+    return new LimestoneApi({
+      ...optsToCopy,
+      arweaveProxy,
+    });
+  }
+
+  async getPrice(
+    tokenSymbol: string,
+    opts: GetPriceOptions = {}): Promise<PriceData | undefined> {
+      const provider = _.defaultTo(opts.provider, this.defaultProvider);
+
+      if (this.useCache) {
+        const price = await this.cacheProxy.getPrice({
+          symbol: tokenSymbol,
+          provider,
+        });
+
+        // Signature verification
+        const shouldVerifySignature = _.defaultTo(
+          opts.verifySignature,
+          this.verifySignature);
+        if (shouldVerifySignature && price !== undefined) {
+          await this.assertValidSignature(price);
+        }
+
+        return price;
+      } else {
+        const txId = await this.arweaveProxy.findTxIdInGraphQL({
+          type: "data",
+          provider,
+          version,
+        });
+
+        if (txId === undefined) {
+          return undefined;
+        }
+
+        const prices = await this.arweaveProxy.getTxDataById(txId, {
+          parseJSON: true,
+        });
+
+        for (const price of prices) {
+          if (price.symbol === tokenSymbol) {
+            return {
+              ...price,
+              provider, // TODO: maybe we want to return provider address here
+              permawebTx: txId,
+            }
+          }
+        }
+      }
+    }
+
+  async getHistoricalPrice(
+    tokenSymbol: string,
+    date: Date,
+    opts: GetPriceOptions = {}): Promise<PriceData | undefined> {
+      if (this.useCache) {
+        const price = await this.cacheProxy.getPrice({
+          symbol: tokenSymbol,
+          provider: _.defaultTo(opts.provider, this.defaultProvider),
+          timestamp: date.getTime(),
+        });
+
+        // Signature verification
+        const shouldVerifySignature = _.defaultTo(
+          opts.verifySignature,
+          this.verifySignature);
+        if (shouldVerifySignature && price !== undefined) {
+          await this.assertValidSignature(price);
+        }
+
+        return price;
+      } else {
+        // TODO: we cannot query ArGQL with timestamp camparators like timestamp_gt
+        // But in future we can think of querying based on block numbers
+        throw new Error(
+          "Fetching historical price from arweave is not supported");
+      }
+  }
+
+  async getHistoricalPrices(
```

Comment:
Let's add support for the interval parameter please. Otherwise, we will need to transfer ~6 prices per day, so rendering a weekly chart will be super slow.